src/util/orm.py

```diff
@@ -289,14 +289,16 @@ def recompute_fields(cr, model, fields, ids=None, logger=_logger, chunk_size=256
 
     if ids is None:
         query = format_query(cr, "SELECT id FROM {} ORDER BY id", table_of_model(cr, model)) if query is None else query
-        cr.execute(query)
-        count = cr.rowcount
-        if count < 2**21:  # avoid the overhead of a named cursor unless we have at least two chunks
+        count_query = "SELECT count(*) FROM ({query})".format(query=query)
+        cr.execute(count_query)
+        count = cr.fetchone()[0]
```

Comment:
Actually, `SELECT count(*)` itself can be slow.
We can just estimate the number of row with `EXPLAIN`
```suggestion
        count_query = format_query(cr, "EXPLAIN (FORMAT JSON) {}", query)
        cr.execute(count_query)
        plan = cr.fetchone()[0]
        count = plan[0]['Plan']['Plan Rows']  # estimated count, actually
```

---
Source: https://wiki.postgresql.org/wiki/Count_estimate

Reply:
It's a nice idea to use the table stats, but I'm unsure we want the logging to be off potentially:
https://github.com/odoo/upgrade-util/pull/322/files#diff-e83e04eeae2ce53657ba28fd8de2126f5f3ef6422a73066af36dcf9191de1278R321-R323

Also, a parallel seq scan that takes 20s should be dwarfed by the overall runtime of this.